### DISCUSSION TITLE
Specify 6d pose (3 for position, 3 for orientation) as target for robot

### DISCUSSION THREAD

**Discussion by Jishnu8** [visose/Robots#215] [Q&A] [answered]
Hi @visose ,
Is it possible to specify the 6d pose (3 for position, 3 for orientation) as the target for the end effector of the robot? Currently, I am aware that we can specify a point along with an appropriate plane as the target for the robot. But is it possible to specify the 6 coordinates directly (x,y,z, roll, pitch, yaw) as the target?

**Comment by visose** [comment, marked as answer]
The Robots plugin requires a `Plane`, so you should convert it to this convention using normal Grasshopper components first.

**Reply by Jishnu8** [reply]
Could you elaborate on how to go about converting the 6d pose to a plane and point in Grasshopper?
Moreover, a plane and a point don't specify a unique representation of the frame of the end effector right? We could have multiple configurations of the end effector frame for which it is both perpendicular to the plane and reaching the desired point.

**Reply by visose** [reply]
In Rhino, a `Plane` object has an origin point + 3 orthonormal vectors. But again, this is unrelated to the Robots plugin and you will have more luck searching in the Rhino forums/documentation.

**Comment by Jishnu8** [comment]
Right. However, am I right in saying that a `Plane` as a target in the robot plugin does not define a unique frame for the end effector? Because we could have multiple configurations of the end effector frame for which the robot attains a `Plane` as a target.

**Reply by visose** [reply]
No, a `Plane` defines an unambiguous position and 3d rotation in space.
